pkg/resource/deploy/source_null.go

```diff
@@ -22,16 +22,19 @@ import (
 	"github.com/pulumi/pulumi/sdk/v3/go/common/util/result"
 )
 
-// NullSource is a singleton source that never returns any resources.  This may be used in scenarios where the "new"
+// NullSource is a source that never returns any resources.  This may be used in scenarios where the "new"
 // version of the world is meant to be empty, either for testing purposes, or removal of an existing stack.
-var NullSource Source = &nullSource{}
```

Comment:
A little worried here since "technically" this is a breaking change if any code happens to depend on `var NullSource`. Perhaps consider leaving that var around? This package is not under `internal` and the identifier is Capitalized so it's exported.

Reply:
I don't think anything uses this, the only reference I can find to NullSource across github.com/pulumi is this use for destroy. Actually even across all github.com search for deploy.NullSource only returns this. We could leave this in as `&nullSource{project: ""}` if we're that worried.

Reply:
I mean something in user code that links against pulumi/pulumi. I'm definitely worried since we have significant codebases linking against us.